jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java

```diff
@@ -34,56 +33,22 @@
 @SuppressWarnings({ "deprecation", "removal" })
 /*[ENDIF] JAVA_SPEC_VERSION >= 17 */
 public final class InternalCRIUSupport {
-	/*[IF CRIU_SUPPORT]*/
-	private static boolean criuSupportEnabled;
-	private static boolean nativeLoaded;
-	private static boolean initComplete;
+	// three states: -1 for unknown, 1 for enabled, 0 for disabled
+	private static int criuSupportEnabled = -1;
```

Comment:
why do we need the unknown state? since the native is in j9vm we can always know

Reply:
`unknown` is for an uninitialized state before `isCRIUSupportEnabledImpl` is invoked, afterwards either `1` - enabled or `0` - disabled, the thought is to save a boolean for `initialized` though each `isCRIUSupportEnabled` performs `criuSupportEnabled == 1` at runtime.
may add it to avoid confusion?

Reply:
assigned `criuSupportEnabled` w/ `isCRIUSupportEnabledImpl()` since first reference of `InternalCRIUSupport` class is expected to be `isCRIUSupportEnabled()`.

@tajila pls have another look.